src/term/transform/apply_args.rs

```diff
@@ -0,0 +1,27 @@
+use crate::term::{Book, Pattern, Term};
+
+impl Book {
+  pub fn apply_args(&mut self, args: Option<Vec<Term>>) -> Result<(), String> {
```

Comment:
We don't want to return a string, but rather an Info on errors. That way we don't lose previous errors.

Check how other passes emit fatal errors.